pkg/xray/resource_xray_license_policy.go

```diff
@@ -20,143 +21,177 @@ func resourceXrayLicensePolicyV2() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the policy (must be unique)",
 			},
 			"description": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "More verbose description of the policy",
 			},
 			"type": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Type of the policy",
+				ValidateDiagFunc: inList("Security", "License"),
 			},
 			"author": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "User, who created the policy",
 			},
 			"created": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Creation timestamp",
 			},
 			"modified": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Modification timestamp",
 			},
 			"rules": {
-				Type:     schema.TypeList,
-				Required: true,
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Nested block describing the policy rules. Described below",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"name": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of the rule",
 						},
 						"priority": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:             schema.TypeInt,
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(1)),
+							Description:      "Integer describing the rule priority. Must be at least 1",
 						},
 						"criteria": {
-							Type:     schema.TypeList,
-							Required: true,
-							MinItems: 1,
-							MaxItems: 1,
+							Type:        schema.TypeSet,
+							Required:    true,
+							Description: "Nested block describing the criteria for the policy. Described below",
+							MinItems:    1,
```

Comment:
Is this still needed?

Reply:
I believe it is. We need to enforce the limitation on certain fields, so users can only submit one element. I've checked all of them, we don't have the limitations where we want to allow several elements.